parl/remote/remote_decorator.py

```diff
@@ -190,33 +194,18 @@ def request_cpu_resource(self, global_client, max_memory):
                     cnt -= 1
                 return None
 
-            def check_attribute(self, attr):
-                '''checkout if attr is a attribute or a function'''
-                self.internal_lock.acquire()
-                self.job_socket.send_multipart(
-                    [remote_constants.CHECK_ATTRIBUTE,
-                     to_byte(attr)])
-                message = self.job_socket.recv_multipart()
-                self.internal_lock.release()
-                tag = message[0]
-                if tag == remote_constants.NORMAL_TAG:
-                    return loads_return(message[1])
-                else:
-                    self.job_shutdown = True
-                    raise NotImplementedError()
-
             def set_remote_attr(self, attr, value):
                 self.internal_lock.acquire()
                 self.job_socket.send_multipart([
-                    remote_constants.SET_ATTRIBUTE,
+                    remote_constants.SET_ATTRIBUTE_TAG,
                     to_byte(attr),
                     dumps_return(value)
                 ])
                 message = self.job_socket.recv_multipart()
                 tag = message[0]
                 self.internal_lock.release()
```

Comment:
we should release the lock after the line `self.remote_attribute_keys_set = loads_return(message[1])`

Reply:
modified